src/Microsoft.TemplateEngine.Core.Contracts/IOrchestrator.cs

```diff
@@ -10,8 +10,19 @@ public interface IOrchestrator
 
         void Run(IGlobalRunSpec spec, IDirectory sourceDir, string targetDir);
 
-        IReadOnlyList<IFileChange2> GetFileChanges(string runSpecPath, IDirectory sourceDir, string targetRoot, string targetDir);
+        IReadOnlyList<IFileChange> GetFileChanges(string runSpecPath, IDirectory sourceDir, string targetDir);
 
-        IReadOnlyList<IFileChange2> GetFileChanges(IGlobalRunSpec spec, IDirectory sourceDir, string targetRoot, string targetDir);
+        IReadOnlyList<IFileChange> GetFileChanges(IGlobalRunSpec spec, IDirectory sourceDir, string targetDir);
+    }
+
+    public interface IOrchestrator2
+    {
+        void Run(string runSpecPath, IDirectory sourceDir, string targetDir);
+
+        void Run(IGlobalRunSpec spec, IDirectory sourceDir, string targetDir);
+
+        IReadOnlyList<IFileChange2> GetFileChanges(string runSpecPath, IDirectory sourceDir, string targetDir);
+
+        IReadOnlyList<IFileChange2> GetFileChanges(IGlobalRunSpec spec, IDirectory sourceDir, string targetDir);
```

Comment:
So the previous commit removed this interface, but we are bringing it back? I was going to suggest default interface implementations again to not have the v2 of this interface (though I'm not sure how applicable to this case).

Reply:
I've reverted all changes done in https://github.com/dotnet/templating/pull/2760 to original state in first commit, as fix was not complete.